### PR TITLE
Permit specifying output file

### PR DESCRIPTION
**Pull request by Forgind**
For get* CLI switches

Fixes #

### Context
Design question:
Should SimpleErrorLogger be enabled if we're redirecting into a file? Should it be enabled whenever we aren't using TerminalLogger?

The reason I ask is that it's the only thing that writes to stderr at the moment, so since I re-enabled normal loggers when writing to a file, that means I disabled SimpleErrorLogger, and now errors come through stdout (with no TerminalLogger) as MSBuild normally does things. I'm not convinced that's ideal...

### Changes Made
Permit using -getResultOutputFile to redirect output from get* into a file. Example:
dotnet msbuild foo.csproj -getProperty:Bar -getResultOutputFile:Biz.txt
This writes the value of property Bar into Biz.txt.

### Testing
~I tried using this, and it failed to load my test project because it was trying to include $(MonoAndroidResourcePrefix)\(various), and MonoAndroidResourcePrefix was not defined. That doesn't feel like it's related to my change, but it's hard for me to firmly state that this works when 100% of my tests have failed 🙃~

I tried again today, and the error mysteriously disappeared. It failed until I added Flush, but now it seems to work properly.

### CODE REVIEW

File: src/MSBuild/XMake.cs [hunk 1]
```diff
@@ -2563,7 +2573,8 @@ private static bool GatherAutoResponseFileSwitchesFromFullPath(string autoRespon
                     getProperty = commandLineSwitches[CommandLineSwitches.ParameterizedSwitch.GetProperty] ?? Array.Empty<string>();
                     getItem = commandLineSwitches[CommandLineSwitches.ParameterizedSwitch.GetItem] ?? Array.Empty<string>();
                     getTargetResult = commandLineSwitches[CommandLineSwitches.ParameterizedSwitch.GetTargetResult] ?? Array.Empty<string>();
-                    if (getProperty.Length > 0 || getItem.Length > 0 || getTargetResult.Length > 0)
+                    getResultOutputFile = commandLineSwitches[CommandLineSwitches.ParameterizedSwitch.GetResultOutputFile].FirstOrDefault() ?? string.Empty;
+                    if ((getProperty.Length > 0 || getItem.Length > 0 || getTargetResult.Length > 0) && getResultOutputFile.Length == 0)
```
Comment (tmds) [inline src/MSBuild/XMake.cs:2577]: suggestion: assign this to a variable, something like `printEvaluationToStdout`.

File: src/MSBuild/XMake.cs [hunk 2]
```diff
@@ -2650,7 +2661,7 @@ private static bool GatherAutoResponseFileSwitchesFromFullPath(string autoRespon
                         commandLineSwitches[CommandLineSwitches.ParameterizedSwitch.BinaryLogger],
                         commandLineSwitches[CommandLineSwitches.ParameterizedSwitch.ProfileEvaluation],
                         groupedFileLoggerParameters,
-                        getProperty.Length + getItem.Length + getTargetResult.Length > 0,
+                        (getProperty.Length + getItem.Length + getTargetResult.Length > 0) && getResultOutputFile.Length == 0,
```
Comment (tmds) [inline src/MSBuild/XMake.cs:2664]: This is the same value, you can use the variable.

File: src/MSBuild/XMake.cs (hunk 1 above)
Comment (Forgind) [reply src/MSBuild/XMake.cs:2577]: It doesn't just control evaluation but also build information. I'm proposing minimizeStdOutOutput. What do you think?

**Comment by tmds** [comment]
Looking forward to have this in an SDK near me!

**Comment by Forgind** [comment]
FYI, I called my commit 'Start on test', but that was just because I don't have the components to build on this laptop, so I didn't run it locally before pushing, but it seems to have built and passed 🙂

File: src/MSBuild.UnitTests/XMake_Tests.cs [hunk 3]
```diff
@@ -809,6 +809,41 @@ public void BuildFailsWithBadPropertyName()
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Bar Include=""ItemContent"" />
+  </ItemGroup>
+
+  <Target Name=""Biz"" />
+</Project>
+");
+            string resultFile = Path.Combine(Path.GetDirectoryName(project.Path), "resultFile.txt");
```
Comment (ladipro) [inline src/MSBuild.UnitTests/XMake_Tests.cs:832]:
```suggestion
            string resultFile = env.GetTempFile(".txt").Path;
```

will delete the file automatically when the test environment is disposed of, so no need for the `try`/`finally` below.

**Review by ladipro — APPROVED** [review]
Thank you!

**Comment by tmds** [comment]
@Forgrind is this a .NET 9 feature only? Or is it expected to become available in a .NET 8 SDK band?

**Comment by rainersigwald** [comment]
@tmds it'll be in MSBuild 17.10/SDK 8.0.3xx.

**Comment by VAllens** [comment]
Good job

**Pull request merged**